pkg/web/static/js/main.js

```diff
@@ -43,4 +43,26 @@ document.body.addEventListener('htmx:responseError', (e) => {
       document.querySelector('.alert').remove()
     }, 5000)
   }
-})
\ No newline at end of file
+})
+
+/*
+ * Timestamp handling: localizes and formats datetimes on the page.
+ */
+const updateDatetimes = () => {
+  const dtfmt = Intl.DateTimeFormat(navigator.languages, {
+    year: 'numeric', month: 'long', day: 'numeric',
+    hour: 'numeric', minute:'numeric', second: 'numeric',
+    hour12: false, timeZoneName: "short",
+  });
+
+  const datetimes = document.querySelectorAll('.datetime');
+  datetimes?.forEach(elem => {
+    if (elem.textContent !== '' || elem.textContent !== null) {
+      const dt = new Date(elem.textContent);
+      elem.textContent = dtfmt.format(dt);
+      elem.classList.remove('datetime');
+    }
+  });
+};
+
+document.body.addEventListener('htmx:afterSettle', updateDatetimes);
```

Comment:
The `DOMContentLoaded` event listener has been removed since all of the timestamps are populated after a swap event.